wordpress-mobile/WordPress-Android
# Issue/9739 domain contact info and registration

Fixes #9739

**[Flux-C counterpart](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1245) should be merged before this PR.**

This PR adds the rest of the domain registration functionality: Contact Details + Privacy form and Domain Registration Result screens.

I spend too much time trying to make it all work with `ViewModels`, but without two-way data binding, the complexity was growing out of hand, so I had to abandon this idea for now, and remade everything with more "traditional" approach. There was some certain element of haste, so please do let me know if I missed something important.

Unfortunately, this PR in a whole is necessary for some subsequent tasks that can not wait, so the size of it is a bit out of control 😓 

To test:
1. Install the wasabi build.
2. Switch to a site on a business plan with unclaimed domain credit (let me know if you need help with this).
3. Navigate to Plugins screen, and from there to details of any plugin (like Woocommerce).
4. Tap on Install button.
5. In the dialog tap on "REGISTER DOMAIN".
6. Pick a domain (preferably .blog).
7. Tap on "CHOOSE DOMAIN" and navigate to contact form details.
8. If you had contact information stored at WordPress.com it should be prefilled.
9. All the fields apart from State and Organisation are required, and their emptiness is validated locally.
10. The validity of input fields content is validated on backend one by one (eg. if you have two fields with malformed you will have to make two registration attempts to get errors for both of them).
11. Make sure ToS link is clickable.
12. Every time you change the country States are loaded from remote (some countries, like Germany, do not have states).
13.  Try to register the domain.

PS: Follow domain testing guideline and make sure if you purchase a domain to cancel it.

I had to make a couple of small functional compromises/changes based on the original design, because of the time constraints. 

![12235](https://user-images.githubusercontent.com/728822/57165155-2808a080-6dab-11e9-82e0-90f7a8e9f927.png)



Update release notes:
